File: data/src/main/java/com/joxad/zikobot/data/module/spotify_api/model/SpotifyAlbum.java

package com.joxad.zikobot.data.module.spotify_api.model;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor(suppressConstructorProperties = true)
public class SpotifyAlbum {

    @SerializedName("id")
    private String id;

    @SerializedName("images")
    public List<Image> images = new ArrayList<Image>();

    @SerializedName("artists")
    public List<SpotifyArtist> artists = new ArrayList<SpotifyArtist>();

    @SerializedName("limit")
    public int limit;

    @SerializedName("total")
    public int total;
    @SerializedName("name")
    public String name;
    @SerializedName("type")
    public String type;
    @SerializedName("album_type")
    protected String albumType;
}

File: data/src/main/java/com/joxad/zikobot/data/module/spotify_api/model/SpotifyAlbum.kt
package com.joxad.zikobot.data.module.spotify_api.model

import com.google.gson.annotations.SerializedName
import java.util.*

class SpotifyAlbum {

    @SerializedName("images")
    var images: List<Image> = ArrayList()
    @SerializedName("artists")
    var artists: List<SpotifyArtist> = ArrayList()
    @SerializedName("limit")
    var limit: Int = 0
    @SerializedName("total")
    var total: Int = 0
    @SerializedName("name")
    var name: String? = null
    @SerializedName("type")
    var type: String? = null
    @SerializedName("album_type")
    var albumType: String? = null
    @SerializedName("id")
    val id: String? = null
}
